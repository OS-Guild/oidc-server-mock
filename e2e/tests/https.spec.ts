import { Agent } from 'https';
import * as dotenv from 'dotenv';
import axios from 'axios';

describe('Https', () => {
  beforeAll(() => {
    dotenv.config();
  });

  test('Discovery Endpoint', async () => {
    const response = await axios.get(process.env.OIDC_DISCOVERY_ENDPOINT_HTTPS, {
      httpsAgent: new Agent({ rejectUnauthorized: false }),
    });
    expect(Object.keys(response.data)).toMatchSnapshot();
  });
});
