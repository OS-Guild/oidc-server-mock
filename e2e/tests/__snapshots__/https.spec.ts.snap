// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Https Discovery Endpoint 1`] = `
Array [
  "issuer",
  "jwks_uri",
  "authorization_endpoint",
  "token_endpoint",
  "userinfo_endpoint",
  "end_session_endpoint",
  "check_session_iframe",
  "revocation_endpoint",
  "introspection_endpoint",
  "device_authorization_endpoint",
  "frontchannel_logout_supported",
  "frontchannel_logout_session_supported",
  "backchannel_logout_supported",
  "backchannel_logout_session_supported",
  "scopes_supported",
  "claims_supported",
  "grant_types_supported",
  "response_types_supported",
  "response_modes_supported",
  "token_endpoint_auth_methods_supported",
  "id_token_signing_alg_values_supported",
  "subject_types_supported",
  "code_challenge_methods_supported",
  "request_parameter_supported",
]
`;
